app/views/datasets/_contact.html.erb

```diff
@@ -1,23 +1,35 @@
 <section class="contact">
   <div>
     <h2 class="heading-medium">
-      <%= t('.contact') %>
+      <%= t('datasets.contact.contact') %>
     </h2>
-    <div class="column-one-half">
-      <h3 class="heading-small">
-      <%= t('.enquiries') %>
-      </h3>
-      <p><%= @dataset.organisation['contact_name'] %> </br>
-        <a href=""><%= @dataset.organisation['contact_email'] %></a></p>
-    </div>
-    <% if @dataset.organisation['foi_name'] || @dataset.organisation['foi_email'] %>
-      <div class="column-one-half foi">
+
+    <% if contact_email_exists?(@dataset) %>
+      <div class="column-one-half">
         <h3 class="heading-small">
-          <%= t('.foi_requests') %>
+          <%= t('datasets.contact.enquiries') %>
         </h3>
-        <p><%= @dataset.organisation['foi_name'] %> </br> <a href=""><%= @dataset.organisation['foi_email'] %></a>
+
+        <p>
+          <%= @dataset.contact_name.presence || @dataset.organisation.contact_name %></br>
+          <%= mail_to(@dataset.contact_email.presence || @dataset.organisation.contact_email) %>
```

Comment:
There is repeated knowledge of which contact and email address to use that is shared amongst the view and helper that could get out of sync.

In the `contact_email_exists` helper method we check if `dataset.contact_email` or `dataset.organisation.contact_email` are present. Then in the view code we duplicate this knowledge with `<%= mail_to(@dataset.contact_email.presence || @dataset.organisation.contact_email) %>`.

We can address that by extracting all the logic into `DatasetsHelper`.

Maybe something like this:

```erb
<%= contact_name_for(dataset) %>
<%= mail_to(contact_email_for(dataset)) %>
```

```rb
def contact_email_exists?(dataset)
  contact_email_for(dataset).present?
end

def contact_name_for(dataset)
  dataset.contact_name || dataset.organisation.contact_name
end

def contact_email_for(dataset)
  dataset.contact_email || dataset.organisation.contact_email
end
```

Also, it's potentially possible that we could display the contact name for the dataset alongside the contact email for the organisation, or the contact name for the organisation and the contact email for the dataset. Is that allowed?

For example:

```
dataset.contact_name = "Joe Bloggs"
dataset.contact_email = "joe.bloggs@example.com"
dataset.organisation.contact_name = "John Doe"
dataset.organisation.contact_email = "john.doe@example.com"
#= Joe Bloggs <joe.bloggs@example.com>

dataset.contact_name = nil
dataset.contact_email = "joe.bloggs@example.com"
dataset.organisation.contact_name = "John Doe"
dataset.organisation.contact_email = "john.doe@example.com"
#= John Doe <joe.bloggs@example.com>

dataset.contact_name = "Joe Bloggs"
dataset.contact_email = nil
dataset.organisation.contact_name = "John Doe"
dataset.organisation.contact_email = "john.doe@example.com"
#= Joe Bloggs <john.doe@example.com>
```

Reply:
The behaviour on legacy is if you add zero contact information, then the dataset displays the organisation contact details. But if you add a single piece of contact information to the dataset, then it displays that single piece, and none of the other organisations contact info.

I recall Laurent discussing this work with Jonathan... perhaps this implementation is to increase the amount of contact information we have on the site of any kind? Or it could just be a mistake. I'll ask Jonathan on the trello ticket.